packages/cli/src/executions/executions.controller.ts

```diff
@@ -21,17 +23,18 @@ export class ExecutionsController {
 		private readonly workflowSharingService: WorkflowSharingService,
 		private readonly activeExecutionService: ActiveExecutionService,
 		private readonly license: License,
+		private readonly workflowRepository: WorkflowRepository,
 	) {}
 
-	private async getAccessibleWorkflowIds(user: User) {
+	private async getAccessibleWorkflowIds(user: User, scope: Scope) {
 		return this.license.isSharingEnabled()
-			? await this.workflowSharingService.getSharedWorkflowIds(user)
-			: await this.workflowSharingService.getSharedWorkflowIds(user, ['workflow:owner']);
```

Comment:
I'm not quite sure what the intention here was:

If sharing is enabled we return all workflows shared with the user.
Otherwise we return all workflows that are shared with the user but only if the user owns the workflows.

----

I changed this to:

If sharing is enabled, return all workflows shared with the user, unless they have global read access, then they get all workflows.
If sharing is not enabled return all workflows regardless. **<- maybe this should instead return only the workflows in the user's personal project?**

Reply:
This behaves different than on master. I'll fix this and add a test.

Reply:
Added a test and fixed it. 

`WorkflowSharingService.getSharedWorkflowIds` can now take scopes and infer the project and workflow roles or the roles can be passed in directly, which makes it possible to only get the workflow ids of workflows that are in the personal project that the user owns, e.g. workflows that have not been shared with them.